elixir/apps/web/lib/web/auth.ex

```diff
@@ -113,17 +123,24 @@ defmodule Web.Auth do
          "nonce" => _nonce,
          "state" => state
        }) do
-    query =
-      %{
-        fragment: encoded_fragment,
-        state: state,
+    client_auth_data =
+      [
         actor_name: identity.actor.name,
-        identity_provider_identifier: identity.provider_identifier
-      }
+        fragment: encoded_fragment,
+        identity_provider_identifier: identity.provider_identifier,
+        state: state
+      ]
       |> Enum.reject(&is_nil(elem(&1, 1)))
 
-    Phoenix.Controller.redirect(conn,
-      to: ~p"/#{conn.assigns.account.slug}/sign_in/success?#{query}"
+    redirect_url = ~p"/#{conn.assigns.account.slug}/sign_in/client_redirect"
+
+    conn
+    |> put_client_auth_data_to_cookie(client_auth_data)
+    |> Phoenix.Controller.put_root_layout(false)
+    |> Phoenix.Controller.put_view(Web.SignInHTML)
+    |> Phoenix.Controller.render("client_redirect.html",
+      redirect_url: redirect_url,
+      layout: false
```

Comment:
Does this duplicate `|> Phoenix.Controller.put_root_layout(false)`? I guess either this line or that one is not needed?

Reply:
In this case, both are needed.  Or at least I believe both are needed.

Since I needed to add a `<meta>` tag to the page, I read that the proper location for this tag is inside the `<head>` tag and not in the `<body>` tag.  I think technically it would work in the `<body>` tag, but I didn't want to risk any strange edge cases here.  So, with that in mind, I didn't see a way to add anything to the `<head>` tag without modifying the root layout and that felt heavy handed in this situation because this is the only place it's needed.  So, what I ended up doing was setting both the root layout and app layout to false and rendering everything from the `sign_in_html.ex` file itself.

If there is a way to get the `<meta>` tag in the head section I'm definitely happy to update things.